custom-metrics-stackdriver-adapter/pkg/adapter/coreprovider/client.go

```diff
@@ -27,53 +28,100 @@ func newClient(translator *translator.Translator) *stackdriverCoreClient {
 	return &stackdriverCoreClient{translator}
 }
 
-// TODO(holubowicz): handle a case when len(resourceNames) > oneOfMax
-func (p *stackdriverCoreClient) getContainerCPU(podsNames []string) (map[string]map[string]resource.Quantity, map[string]api.TimeInfo, error) {
-	stackdriverRequest, err := p.translator.GetSDReqForContainersWithNames(podsNames, containerCPUMetricName, metricKindCPU, labels.Everything(), translator.AllNamespaces)
-	if err != nil {
-		return nil, nil, err
+func (p *stackdriverCoreClient) getPodMetric(podsNames []string, metricName string, metricKind string, labels labels.Selector) (map[string]map[string]resource.Quantity, map[string]api.TimeInfo, error) {
+	numOfRequests := (len(podsNames) + translator.MaxNumOfArgsInOneOfFilter - 1) / translator.MaxNumOfArgsInOneOfFilter // ceil
```

Comment:
I think we should restrict maximum number of pararel requests. e.g. setup pool of 5 goroutines that read from channel